.github/workflows/terraform-deploy-template.yml

```diff
@@ -0,0 +1,73 @@
+name: Terraform_deploy_workflow
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+      working_directory:
+        type: string
+        required: true
+      tf_vars:
+        type: string
+        required: true
+      backend_config:
+        type: string
+        required: true
+    secrets:
+      azure_ad_client_id:
+        required: true
+      azure_ad_client_secret:
+        required: true
+      azure_subscription_id:
+        required: true
+      azure_ad_tenant_id:
+        required: true
+        
+jobs:
+  terraform:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    name: Terraform
+    # Use the Bash shell regardless whether the GitHub Actions runner is ubuntu-latest, macos-latest, or windows-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.working_directory }}
+        shell: bash
+    env:
+      -template: ../../environments/dev.tfvars
+      ARM_CLIENT_ID: ${{ secrets.azure_ad_client_id }}
+      ARM_CLIENT_SECRET: ${{ secrets.azure_ad_client_secret }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.azure_subscription_id }}
+      ARM_TENANT_ID: ${{ secrets.azure_ad_tenant_id }}
+      tf_vars: ${{ inputs.tf_vars }}
+      backend_config: ${{ inputs.backend_config }}
+
+    steps:
+    # Checkout the repository to the GitHub Actions runner
+    - name: Checkout
+      uses: actions/checkout@v3
+    
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v1
```

Comment:
There's a version 2 of this now.  I don't know what that gets you - but might be worth trying that out?